src/main/java/com/ahsoka/SALC/user_model/persistance/entity/User.java

```diff
@@ -0,0 +1,65 @@
+package com.ahsoka.SALC.user_model.persistance.entity;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.springframework.data.elasticsearch.annotations.Document;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.util.Collection;
+import java.util.Collections;
+
+@Getter
+@Setter
+@EqualsAndHashCode
+@NoArgsConstructor
+@Document(indexName = "users", shards = 1, createIndex = false)
+public class User implements UserDetails {
+
+    private String id;
+    private String email;
+    private String password;
+    private Role role;
+    private final boolean locked = false;
+    private final boolean enabled = true;
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+
+        SimpleGrantedAuthority authority = new SimpleGrantedAuthority(role.getName());
```

Comment:
Tras mirar en [Enum Java 14](https://docs.oracle.com/en/java/javase/14/docs/api/java.base/java/lang/Enum.html), el método quedaría:

```suggestion
SimpleGrantedAuthority authority = new SimpleGrantedAuthority(role.name());
return Collections.singletonList(authority);
```